microsoft/PowerToys
# [Settings/Run] LowLevel Keyboard hooking for Hotkeys

## Summary of the Pull Request
Replaced use of `NHotkey.Wpf` with low level keyboard hooks for both Settings and Run.

 
## References


## PR Checklist
* [x] Applies to #2325


## Detailed Description of the Pull Request / Additional comments
* Removed old Hotkey code from Run, including dependencies to the `NHotkey` nuget package.
* Introduced new classes to the `interop` library, which can be used to register callbacks for Key Presses using low level hooking.


## Validation Steps Performed
1. Start PowerToys.
2. Go to Settings\Run.
3. Change the Hotkey to start the Run prompt (for example, to Win + Space)
4. Wait a few seconds and use the hotkey.
5. The Run prompt should appear.
